legend-engine-test-runner-service/src/main/java/org/finos/legend/engine/testable/service/connection/TestConnectionBuilder.java

```diff
@@ -65,7 +65,13 @@ public Pair<Connection, List<Closeable>> visit(Connection connection)
                 .select(Optional::isPresent)
                 .collect(Optional::get)
                 .getFirstOptional()
-                .orElseThrow(() -> new UnsupportedOperationException("Unsupported embedded data type '" + embeddedData.getClass().getSimpleName() + "\" with connection type '" + connection.getClass().getSimpleName() + '"'));
+                .orElseThrow(() ->
+                {
+                    String errorMessage = embeddedData == null
+                            ? "No test data provided for connection type '" + connection.getClass().getSimpleName() + "\". Either you need to provide test data for the connection type or connection type is not supported."
```

Comment:
The quoting of the type name is incorrect. You have a opening single quote (before connection.getClass()) but closing double quote ?

Reply:
Thanks for pointing this. Done